pkg/fleet/internal/cdn/config.go

```diff
@@ -16,33 +19,85 @@ const (
 	doNotEditDisclaimer = `# This configuration was generated by Datadog's Fleet Automation. DO NOT EDIT.`
 )
 
-// config is a configuration for the package.
-type config map[interface{}]interface{}
-
 // layer is a config layer that can be merged with other layers into a config.
 type layer struct {
-	ID      string
-	Content map[interface{}]interface{}
+	ID                  string                 `json:"name"`
+	AgentConfig         map[string]interface{} `json:"config"`
+	SecurityAgentConfig map[string]interface{} `json:"security_agent"`
+	SystemProbeConfig   map[string]interface{} `json:"system_probe"`
 }
 
 // newConfig creates a new config from a list of layers.
-func newConfig(layers ...layer) (config, error) {
-	config := make(map[interface{}]interface{})
-	var layerIDs []string
+func newConfig(layers ...*layer) (_ *Config, err error) {
+	layerIDs := []string{}
+	mergedLayer := &layer{
+		AgentConfig:         map[string]interface{}{},
+		SecurityAgentConfig: map[string]interface{}{},
+		SystemProbeConfig:   map[string]interface{}{},
+	}
+
+	// Merge all layers in order
 	for _, l := range layers {
-		merged, err := merge(config, l.Content)
-		if err != nil {
-			return nil, err
-		}
-		config = merged.(map[interface{}]interface{})
 		layerIDs = append(layerIDs, l.ID)
+		if l.AgentConfig != nil {
+			agentConfig, err := merge(mergedLayer.AgentConfig, l.AgentConfig)
+			if err != nil {
+				return nil, err
+			}
+			mergedLayer.AgentConfig = agentConfig.(map[string]interface{})
+		}
+
+		if l.SecurityAgentConfig != nil {
+			securityAgentConfig, err := merge(mergedLayer.SecurityAgentConfig, l.SecurityAgentConfig)
+			if err != nil {
+				return nil, err
+			}
+			mergedLayer.SecurityAgentConfig = securityAgentConfig.(map[string]interface{})
+		}
+
+		if l.SystemProbeConfig != nil {
+			systemProbeAgentConfig, err := merge(mergedLayer.SystemProbeConfig, l.SystemProbeConfig)
+			if err != nil {
+				return nil, err
+			}
+			mergedLayer.SystemProbeConfig = systemProbeAgentConfig.(map[string]interface{})
+		}
+	}
+	mergedLayer.AgentConfig[layerKeys] = layerIDs
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Is this to report the applied layers? Let's add a description around it